### ISSUE TITLE
Request: Official uv devcontainer

### ISSUE THREAD

**Issue by rdong8** [astral-sh/uv#8737]
Would be really cool if we could get an official `uv` devcontainer for remote development using common IDEs (VS Code, PyCharm). It isn't too difficult to install `uv` in the Dockerfile by hand but an official option would be great. [Currently](https://github.com/devcontainers/images/tree/main/src/python) the Python devcontainers are published version-by-version, so we don't have access to 3.13 yet.

**Comment by wstrausser** [comment]
Agreed that this would be great to have. For now, I've just been using the following Dockerfile for uv-based devcontainers:

```
FROM mcr.microsoft.com/devcontainers/base:bullseye

USER vscode
RUN curl -LsSf https://astral.sh/uv/install.sh | sh

```

**Comment by ryanhiebert** [comment]
If you want to avoid a custom dockerfile, you might be able to use this feature from the `devcontainer.json`. It's been working pretty well for me.

```json
{
    "features": {
        "ghcr.io/va-h/devcontainers-features/uv:1": {}
    }
}
```

**Comment by tobiasdiez** [comment]
At https://github.com/tobiasdiez/sage/tree/fedora-devcontainer-meson/.devcontainer/uv a working devcontainer feature for uv can be found. If someone wants to continue working on this, please feel free to use it as a starting point. (Essentially only publishing it to the ghcr repository is missing)

**Comment by v-fidelusaleksander** [comment]
Looks like it was added to the `devcontainers-extra` repository recently (September 2025) 

https://github.com/devcontainers-extra/features/tree/main/src/uv

**Comment by coretl** [comment]
Another alternative (for those that find features make the devcontainer build too slow), we added uv (and other build tools) to a base ubuntu container, then use `uv` to manage the python version, and publish in step with upstream ubuntu:
https://github.com/DiamondLightSource/ubuntu-devcontainer

**Comment by dunnkers** [comment]
There's also [https://github.com/dunnkers/python-uv-devcontainer](https://github.com/dunnkers/python-uv-devcontainer)


> Disclaimer: I created that repo. It's using a [Devcontainer Features](https://containers.dev/features) so there's little extra install scripts necessary to get it to work.

**Comment by ryan-minato** [comment]
Support this request +1. However, we need to address a major blocker: broken symlinks in `.venv` when container restarts.

In Dev Container environments, `uv` defaults to symlinking the Python interpreter. Since the container's root filesystem is often ephemeral while the workspace is persisted, these links break upon restart. Even with `UV_LINK_MODE=copy`, the interpreter remains a symlink, forcing `uv` to delete and full environment re-installation.

This is a huge issue for ML projects (which need pytorch, flash-attention) where re-installation is slow. I've detailed this in #17754 .

**Comment by ryan-minato** [comment]
An Image or Feature alone cannot fix this, as volume mounting is managed by `devcontainer.json`. My current workaround is to manually mount a volume for `~/.cache/uv` and use a `postCreateCommand` to fix ownership, but this is not a sustainable solution.

**Comment by coretl** [comment]
> An Image or Feature alone cannot fix this, as volume mounting is managed by `devcontainer.json`. My current workaround is to manually mount a volume for `~/.cache/uv` and use a `postCreateCommand` to fix ownership, but this is not a sustainable solution.

Our solution was to put the uv cache, the uv managed python and the created project venv all in the same volume mount:
https://github.com/DiamondLightSource/python-copier-template/blob/main/.devcontainer/devcontainer.json

**Comment by itanex** [comment]
I have been looking to update my `.devcontainer` file to include this. However, the IT department I work with has not unblocked these specific third party solutions for us to use. Is there a timeline to bring the "dev container" `feature` support to the distribution? Or am I and my team stuck with managing an actual `dockerfile` definition to get `uv` into the developer environments?
